Should not crash if a package is not found

Please file this bug report at https://github.com/marius-h/flutter_enhancement_suite/issues/new.

---

## What happened

Hi, thanks for this great plugin!

It seems like an internal package that is not published on PubDev named flutter_kvk is causing this plugin to crash.

## Version information

IntelliJ IDEA `2023.1.5` • Flutter plugin `io.flutter 75.1.3` • Dart plugin `231.9392.3`

Flutter 3.13.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 2524052335 (3 weeks ago) • 2023-09-06 14:32:31 -0700
Engine • revision b8d35810e9
Tools • Dart 3.1.1 • DevTools 2.25.0

## Exception

ExternalToolPass:

```
com.intellij.diagnostic.PluginException: annotator: de.mariushoefler.flutterenhancementsuite.inspections.PackageUpdateExternalAnnotator@57c6c9e6 (class de.mariushoefler.flutterenhancementsuite.inspections.PackageUpdateExternalAnnotator) [Plugin: de.mariushoefler.flutter_enhancement_suite]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:89)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.processError(ExternalToolPass.java:259)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:207)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:198)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.runChangeAware(ExternalToolPass.java:274)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$1(ExternalToolPass.java:176)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:361)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:339)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:175)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:350)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:340)
	at com.intellij.util.ui.update.MergingUpdateQueue.doExecute(MergingUpdateQueue.java:297)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:280)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:245)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:249)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:365)
	at com.intellij.util.Alarm$Request.run(Alarm.java:354)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:272)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:248)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:28)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:221)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:227)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:210)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: de.mariushoefler.flutterenhancementsuite.exceptions.GetLatestPackageVersionException: Cannot get the latest version number for package: flutter_kvk
	at de.mariushoefler.flutterenhancementsuite.utils.PubApi.getPackageLatestVersion(PubApi.kt:86)
	at de.mariushoefler.flutterenhancementsuite.utils.FileParser.mapToVersionDescription(FileParser.kt:64)
	at de.mariushoefler.flutterenhancementsuite.utils.FileParser.getVersionsFromFile(FileParser.kt:21)
	at de.mariushoefler.flutterenhancementsuite.utils.FileParser.checkFile(FileParser.kt:11)
	at de.mariushoefler.flutterenhancementsuite.inspections.PackageUpdateExternalAnnotator.doAnnotate(PackageUpdateExternalAnnotator.kt:25)
	at de.mariushoefler.flutterenhancementsuite.inspections.PackageUpdateExternalAnnotator.doAnnotate(PackageUpdateExternalAnnotator.kt:19)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:204)
	... 36 more
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+### Fixed
+- Fix exception when including a package which is not published on pub.dev (#260)
+
 ## 1.7.0 - 2023-09-17
 
 ### Added

--- a/src/main/kotlin/de/mariushoefler/flutterenhancementsuite/utils/FileParser.kt
+++ b/src/main/kotlin/de/mariushoefler/flutterenhancementsuite/utils/FileParser.kt
@@ -18,7 +18,7 @@ class FileParser(private val file: PsiFile) {
         val problemDescriptionList = mutableListOf<VersionDescription>()
         val packageLines = file.readPackageLines()
         packageLines
-            .map { pair -> mapToVersionDescription(pair) }
+            .mapNotNull { pair -> mapToVersionDescription(pair) }
             .forEach { versionDescription ->
                 val currentVersionWithoutExtras = versionDescription.currentVersion.split("-").first()
                 if (isNewerVersion(currentVersionWithoutExtras, versionDescription.latestVersion)) {
@@ -53,12 +53,15 @@ class FileParser(private val file: PsiFile) {
         return false
     }
 
-    @Throws(GetLatestPackageVersionException::class)
-    private fun mapToVersionDescription(it: Pair<String, Int>): VersionDescription {
+    private fun mapToVersionDescription(it: Pair<String, Int>): VersionDescription? {
         val dependency = it.first
         val counter = it.second
 
-        val latestVersion = PubApi.getPackageLatestVersion(dependency.getPubPackageName())
+        val latestVersion = try {
+            PubApi.getPackageLatestVersion(dependency.getPubPackageName())
+        } catch (e: GetLatestPackageVersionException) {
+            return null
+        }
         val currentVersion = dependency.getCurrentPubPackageVersion()
 
         return VersionDescription(counter, currentVersion, latestVersion)

